### PR TITLE
Bug 1333465, added SSH topic to Dev Guide

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1333465

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@adellape @tnguyen-rh @tpoitras @bfallonf PTAL

File: dev_guide/ssh_environment.adoc [hunk 1]
```diff
@@ -0,0 +1,37 @@
+= Managing Applications in a Secure Shell Environment
```
Comment (tnguyen-rh) [inline dev_guide/ssh_environment.adoc]: This is singular "an SSH env" while the `_build_cfg.yml` Name is plural.  Suggest changing one to be consistent w/ the other.

File: dev_guide/ssh_environment.adoc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Open a remote shell session to a container:
+----
+$ oc rsh <pod_ID>
+----
+
+[[ssh-environment-secure-shell-session-help]]
+== Secure Shell Session Help
+
+For help with usage, options, and to see examples, use the following command.
```
Comment (tnguyen-rh) [inline dev_guide/ssh_environment.adoc]: Probably you can s/, use the following comamnd./:/

File: dev_guide/ssh_environment.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+Managing applications in a secure shell (SSH) environment provides specialized
```
Comment (tnguyen-rh) [inline dev_guide/ssh_environment.adoc]: This boils down to "Managing provides tools", which i don't understand.

**Comment by tnguyen-rh** [comment]
@ahardin-rh One fuzziness and a couple nits.

**Comment by bfallonf** [comment]
@ahardin-rh I think this could do with a bit of an explanation on what you can do once you've ssh'ed into the container. You mention advanced operations and general debugging. Could you give examples of this?

Also, is there a difference between this process and what's outlined in [executing remote commands](https://docs.openshift.com/enterprise/3.2/dev_guide/executing_remote_commands.html)? Is ssh'ing about getting into the container and staying in there, but executing remote commands a one-time thing?

File: dev_guide/ssh_environment.adoc (hunk 3 above)
Comment (ahardin-rh) [inline dev_guide/ssh_environment.adoc]: @tnguyen-rh, you're right and I keep beating my head off of the desk trying to figure out a better way to rewrite this. Thinking on it...

**Comment by ahardin-rh** [comment]
@tnguyen-rh Reworked this topic based on our conversation this morning and your most excellent pointers. PTAL and let me know what you think. Thanks for all of your help!

File: dev_guide/ssh_environment.adoc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+----
+$ oc rsh <pod_ID>
+----
+
+While in the remote shell, you can issue commands as if you are inside the
+container and perform local operations like monitoring, debugging, and using CLI
+commands specific to what is running in the container.
+
+For example, in a MySQL container, you can use `SELECT COUNT([column]) FROM
+[table];` to count the number of records in the database. You can also use use
```
Comment (tnguyen-rh) [inline dev_guide/ssh_environment.adoc]: That `SELECT` command is pretty cool.  Unfortunately, it's not a command that you can type directly into the shell; the actual shell command is called `mysql`.  You invoke that command to get another prompt, and then type in the `SELECT` command at that prompt.  Poking around, i see that we already have an example documented -- see `using_images/db_images/mysql.adoc` around line 106.  Perhaps you can marry this `SELECT` command w/ that example (copying it here).  Another idea is to link to it.

**Comment by tnguyen-rh** [comment]
@ahardin-rh I found only one weirdness.  Apart from that, LGTM! :+1:

**Comment by ahardin-rh** [comment]
@tnguyen-rh Thanks! Fixed!

**Comment by bfallonf** [comment]
@ahardin-rh As discussed, this looks good! I give it a thumbs up :+1:

**Comment by luciddreamz** [comment]
This looks good. At some point it might be worth mentioning that users should SSH into their pods and use the `env` command to list all available environment variables, but we can probably mention that in another section.

**Comment by xingxingxia** [comment]
As QA of the bug, examined the PR, LGTM except:
1. An extra 's' in "make a changes"
2. Better to write as "oc rsh <pod>" instead of "oc rsh <pod_ID>". I suggest this because several days ago someone asks me about the info in `oc delete -h` which uses "ID" in many places but actually `oc delete pod $uid` outputs:  pods "9fddc973-2b8d-11e6-9576-0e3d364e19a5" not found. (Where $uid comes from oc get pod <pod> -o yaml | grep uid). So "ID" seems a bit of confusing. (When you have time, could update this issue in other docs: such as https://docs.openshift.org/latest/cli_reference/basic_cli_operations.html which tells users to use "ID")
Thank you!

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../dev_guide/ssh_environment.adoc[Opening a Remote Shell to Containers]
|Added a new topic on opening a remote shell to containers.
%

**Pull request merged**